### PR TITLE
Added ability to edit tokens

### PR DESCRIPTION
**Pull request by JamesSmartCell**
Edit tokens on a long press of the token on token screen.

### CODE REVIEW

**Comment by michaelScoff** [comment]
@James-Sangalli haha, this is funny! I think @madcake has been working on this feature yesterday. Can you take a look?

**Comment by JamesSmartCell** [comment]
Hi, this is just us trying to figure out your architecture - I was supposed to be submitting the pull request to our fork but you can have the update if you want it.

File: app/src/main/java/com/wallet/crypto/trustapp/interact/AddTokenInteract.java [hunk 1]
```diff
@@ -23,4 +23,12 @@ public Completable add(String address, String symbol, int decimals) {
                         .addToken(wallet, address, symbol, decimals)
                         .observeOn(AndroidSchedulers.mainThread()));
     }
+
+    public Completable update(String address, String symbol, int decimals) {
```
Comment (madcake) [inline app/src/main/java/com/wallet/crypto/trustapp/interact/AddTokenInteract.java]: It is necessary either to rename the class or to move to other class update method.

File: app/src/main/java/com/wallet/crypto/trustapp/repository/RealmTokenSource.java [hunk 2]
```diff
@@ -84,7 +90,35 @@ private void putInNeed(NetworkInfo networkInfo, Wallet wallet, TokenInfo tokenIn
...
+    private void updateInNeed(NetworkInfo networkInfo, Wallet wallet, TokenInfo tokenInfo) {
+        Realm realm = null;
+        try {
+            realm = getRealmInstance(networkInfo, wallet);
+            RealmTokenInfo realmTokenInfo = realm.where(RealmTokenInfo.class)
+                    .equalTo("address", tokenInfo.address)
+                    .findFirst();
+            if (realmTokenInfo != null) {
+                realm.executeTransaction(r ->
+                {
```
Comment (madcake) [inline app/src/main/java/com/wallet/crypto/trustapp/repository/RealmTokenSource.java]: Format as everywhere else.

File: app/src/main/java/com/wallet/crypto/trustapp/repository/RealmTokenSource.java [hunk 3]
```diff
@@ -84,7 +90,35 @@ private void putInNeed(NetworkInfo networkInfo, Wallet wallet, TokenInfo tokenIn
...
-        } finally {
+        }
+        finally {
+            if (realm != null) {
+                realm.close();
+            }
+        }
+    }
+
+    private void updateInNeed(NetworkInfo networkInfo, Wallet wallet, TokenInfo tokenInfo) {
```
Comment (madcake) [inline app/src/main/java/com/wallet/crypto/trustapp/repository/RealmTokenSource.java]: Is this method necessary?

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -65,6 +65,11 @@
             android:label="@string/title_add_token"
             />
 
+        <activity
+            android:name=".ui.EditTokenActivity"
```
Comment (madcake) [inline app/src/main/AndroidManifest.xml]: It is better to extend and modify AddTokenActivity.

File: app/src/main/java/com/wallet/crypto/trustapp/router/EditTokenRouter.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import com.wallet.crypto.trustapp.C;
+import com.wallet.crypto.trustapp.ui.EditTokenActivity;
+
+/**
+ * Created by James on 12/01/2018.
+ */
+
+public class EditTokenRouter
+{
+    public void open(Context context, String address, String symbol, int decimals) {
```
Comment (madcake) [inline app/src/main/java/com/wallet/crypto/trustapp/router/EditTokenRouter.java]: It's better to send TokenInfo or just an address.

**Comment by JamesSmartCell** [comment]
Hi Madcake - yes agree on all points! I wasn't really ready to try to push to your main branch, was planning on refactoring before any main branch push. Thankyou for the follow up.

Cheers, James

**Comment by JamesSmartCell** [comment]
I'm closing this as the functionality has been superceeded by the newer request (no need to ever edit a token if you pull the info in from the actual contract)

**Pull request closed without merging**